sdk/nodejs/string.ts

```diff
@@ -0,0 +1,28 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Output, output } from "./resource";
+
+export function string(strings: TemplateStringsArray, ...args: any[]): Output<string> {
+    return output(args).apply(resolved => {
+        let acc: string = "";
+        for (let i = 0; i < strings.length; i++) {
+            acc += strings[i];
+            if (i < args.length) {
+                acc += `${resolved[i]}`;
```

Comment:
i'm a bit confused by the length checking, only because i''m trying to figure out how you could ever get something that is out of bounds.  is this just be super defensie, or is this for a case that can actually arise?

Reply:
Consider the following:
```
const foo = pulumi.string`pre ${v} post`;
```

I would naively expect there to be two strings and one arg. It's possible that a second arg is synthesized s.t. `strings.length` is always equal to `args.length`, but I'd need to read the spec.

Reply:
What i've found online is:

> The number of template strings is always one plus the number of substitutions. In other words: every substitution is always surrounded by two template strings.

So this makes sense.  thanks!